modules/weapon_helper/functions/fn_addWeapon.sqf

```diff
@@ -0,0 +1,100 @@
+/*
+ * Author: TinfoilHate
+ * Adds a random weapon and appropriate random magazines from an array, either to the unit in general or a specific container on the unit.
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Unit's Olsen Gear Type <STRING>
+ * 2: Array of Weapon/Magazine/Accessory classes <ARRAY>
+ * 3: Optional location to fill with magazines ("" or blank is all locations) <STRING>
+ * 4: Optional maximum magazines to add <NUMBER>
+ *
+ * Return Value:
+ * Nil
+ *
+ * Example:
+ * [_unit, _type, ["rhs_weap_akm", "rhs_30Rnd_762x39mm", "rhs_weap_ak74n", "rhs_30Rnd_545x39_AK"], "vest", 8] call kobld_weaponHelper_fnc_addWeapon;
+ *
+ * Public: Yes
+*/
+
+params ["_unit", "_type", "_inputArray", ["_location", ""], "_max"];
+
+private _accBase = [];
+private _weaponBase = [];
+private _magazinesBase = [];
+
+// Sorting input array
+{
+	_item = toLower(_x);
+
+	if (([(configFile >> "CfgWeapons" >> _item),true] call BIS_fnc_returnParents) findIf {_x == "ItemCore"} != -1) then {
```

Comment:
These can be nested otherwise they continue to iterate through the other checks.
eg. _accBase item still checks for inclusion in _weaponBase and _magazinesBase